.github/workflows/actions/resolve-go-channels/action.yaml

```
@@ -0,0 +1,160 @@
+name: Resolve Go snap channels
+description: |
+  Compiles a list of Go snap channels with unique versions according to the
+  given optional input flags and specific channels. Looks up the snapd build
+  and snapd FIPS build Go channels from build-aux/snap/snapcraft.yaml. Assumes
+  risk stable if not specified.
+
+inputs:
+  include-snapd-build-go-channel:
+    description: Flag instructing to include the channel of Go snap used to build Snapd snap
+    required: false
+    type: boolean
+  include-snapd-build-fips-go-channel:
+    description: Flag instructing to include the channel of Go snap used to build Snapd snap for FIPS
+    required: false
+    type: string
+  include-latest-go-channel: 
+    description: Flag instructing to include the latest channel of Go snap
+    required: false
+    type: boolean
+  specific-go-channels:
+    description: Space separated list of required Go snap channels
+    required: false
+    type: string     
+
+outputs:
+  go-channels:
+    description: JSON list of Go snap channels
+    value: ${{ steps.resolve-go-channels.outputs.go-channels }}
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve Go snap channels
+      id: resolve-go-channels
+      shell: bash
+      run: |
+        # Get the Go snap version corresponding to a channel format <version>[/<risk>]
+        # - If optional risk is omitted, default stable will be assumed
+        # - Assumes both device and snap architecture amd64
+        go_version_from_channel() {
+          channel=$1
+          risk_default="stable"
+          arch_default="amd64"
+
+          # channel=<track>/<risk>
+          if [[ "$channel" =~ ^([0-9]+\.[0-9]+|[0-9]+\.[0-9]+-fips|latest)/(stable|candidate|beta|edge)$ ]]; then
+            track=${channel%%/*}
+            risk=${channel##*/}
+          # channel=<track>
+          elif [[ "$channel" =~ ^([0-9]+\.[0-9]+|[0-9]+\.[0-9]+-fips|latest)$ ]]; then
+            track=$channel
+            risk=$risk_default
+          # Not supported
+          else
+            echo "Cannot use Go channel \"$channel\""
+            return 1
+          fi
+
+          # Query params
+          device_arch="Snap-Device-Architecture: $arch_default"
+          channel_arch="$arch_default"
+          device_series="Snap-Device-Series: 16"
+          content_type="Content-Type: application/json"
+          endpoint="https://api.snapcraft.io/v2/snaps/info/go"
+
+          # Query store
+          if ! result=$(curl -s -H "$device_arch" -H "$device_series" -X GET -H "$content_type" "$endpoint"); then
```

```suggestion
          if ! result=$(curl -s --fail -H "$device_arch" -H "$device_series" -X GET -H "$content_type" "$endpoint"); then
```
otherwise 4xx, 5xx will not be treated as errors

---

Nice improvement, thanks. Fixed